### PR TITLE
NETOBSERV-1379: enhance DNS debugging to dbg DNS over TCP with NA fields

### PR DESCRIPTION
**Pull request by msherif1234**
## Description

it was noticed with DNS over TCP that the handshake packets are too small and it doesn't contain any DNS data from UI DNS fields showed as `n/a` which is true

this PR propagate the DNS lookup return code so we can differentiate between cases like TCP handshake case and real errors enriching with DNS info.

when we get N/A `bpf_skb_load_bytes()` return an error
```golang
working
======
 "dns_record": {
                        "id": 40514,
                        "flags": 34176,
                        "latency": 185794,
                        "errno": 0,
                        "offset": 68,
                        "tcp_len": 34,
                        "skb_len": 291
                    },

none working
==========
 "dns_record": {
                        "id": 0,
                        "flags": 0,
                        "latency": 0,
                        "errno": 7,
                        "offset": 68,
                        "tcp_len": 34,
                        "skb_len": 66
                    },

```
in the none working case the received packet is very small that is why we can't fetch DNS header info.

Those small packets are TCP handshake and those packets don't have data by definition this clear when u look at the pcap

image (1)

## Dependencies

https://github.com/netobserv/network-observability-console-plugin/pull/425
https://github.com/netobserv/flowlogs-pipeline/pull/533

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [ ] Will this change affect NetObserv / Network Observability operator? If not, you can ignore the rest of this checklist.
* [ ] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [x] Does this PR require product documentation?
  * [x] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [x] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [x] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [ ] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

**Comment by msherif1234** [comment]
when we get N/A bpf_skb_load_bytes() return an error
```sh
Not working
==========
"dns_record": {
                        "id": 0,
                        "flags": 0,
                        "latency": 0,
                        "errno": 14 <<<<<<<<<<<<<<< "EFAULT"
                    },

working
=======
dns_record": {
                        "id": 31706,
                        "flags": 34176,
                        "latency": 228919,
                        "errno": 0
                    },
```

**Comment by jpinsonneau** [comment]
@msherif1234 are there error codes we should expect ?
https://elixir.bootlin.com/linux/v4.7/source/include/uapi/asm-generic/errno-base.h#L10

**Comment by msherif1234** [comment]
> @msherif1234 are these error codes we should expect ? https://elixir.bootlin.com/linux/v4.7/source/include/uapi/asm-generic/errno-base.h#L10

Correct

File: pkg/decode/decode_protobuf.go [hunk 1]
```diff
@@ -90,11 +90,13 @@ func PBFlowToMap(flow *pbflow.Record) config.GenericMap {
 		}
 	}
 
+	out["DnsErrno"] = flow.GetDnsErrno()
```
Comment (jpinsonneau) [inline pkg/decode/decode_protobuf.go:93]: As discussed, my concern here is adding unecessary field on non dns related flows.
We agreed to keep it as is for now.

**Review by jpinsonneau — APPROVED** [review]
LGTM, just a note for future improvment.
Thanks @msherif1234 !

**Pull request merged**